### ISSUE TITLE
ES6 Modules Broken

### ISSUE THREAD

**Issue by ghost** [bjornharrtell/jsts#311]
I get the following error when trying to use ES6 modules in conjunction with `GeometryFactory.createLinearRing`:

```
LineString.js:20 Uncaught TypeError: this.init is not a function
    at LinearRing.LineString (LineString.js:20)
    at new LinearRing (LinearRing.js:18)
    at GeometryFactory.createLinearRing (GeometryFactory.js:185)
    at GeometryFactory.createLinearRing (GeometryFactory.js:182)
    ...
```

Here's a code sample to reproduce the error:
```
import GeometryFactory from 'jsts/org/locationtech/jts/geom/GeometryFactory';

const geometryFactory = new GeometryFactory();
geometryFactory.createLinearRing([]);
```

I notice you included [topolis](https://github.com/bjornharrtell/topolis/blob/master/src/utils.js) as an example of a library relying on the ES6 modules. I looked at their code and they only use `GeometryFactory.createLineString` which does indeed work. I suspect the issue lies in the prototype chain of `LinearRing` since it's 3 levels deep. I.e. `LinearRing` -> `LineString` -> `Geometry`.

In fact, if I manually merge all the `LineString` functions into `LinearRing`, everything works as expected.

I'm using version 1.4.0.

**Comment by bjornharrtell** [comment]
Can't understand why this would affect ES6 modules only and not sure about why the inheritance doesn't work here either. Will have to investigate a bit when possible, would like to avoid the hacked fix.

**Comment by DenisCarriere** [comment]
@bjornharrtell The `index.js` is empty in your published npm repo.

https://unpkg.com/jsts@1.4.0/

![image](https://user-images.githubusercontent.com/550895/30776497-725f2cd0-a075-11e7-8f7c-6295c060417a.png)

Why don't you include your `src` folder in your npm bundle and simply add `"module": "src/jsts.js"` to support ES6 modules.

**Comment by bjornharrtell** [comment]
Because jsts.js is intended for the ES5 bundle. When using  ES6 modules directly I import modules individually. The reason is that re-exporting * imports like in jsts.js defeats tree shaking.

What issues does an empty index.js give you?

**Comment by bjornharrtell** [comment]
To clarify, by making a package with /src/* moved to /* I can import individual modules like in the following example:

```js
import Coordinate from 'jsts/org/locationtech/jts/geom/Coordinate'
```

I.e there is no single point of entry to the library in the ES6 module version.

**Comment by DenisCarriere** [comment]
>  jsts.js is intended for the ES5 bundle

This makes more sense now.

> What issues does an empty index.js give you?

If the `package.json` includes `"module": "index.js"` then bundlers like Rollup will use that file instead of the `"main": "dist/jsts.min.js"` and since it's an empty file it will indicate that there's no exported members.

If there's no intent for ES6 modules, then `jsts` should remove `"module":"index.js"` from the `package.json`.

**Comment by bjornharrtell** [comment]
There is an intent and that is that while there are no default exported module(s) a dependent project can import individual modules i.e from a ES6 perspective JSTS is a library/collection of modules with no entry point.

An example where I depend on JSTS and import individual modules is my own project https://github.com/bjornharrtell/topolis, see imports in https://github.com/bjornharrtell/topolis/blob/master/src/utils.js. I've used it in similar ways for other non open projects.

**Comment by bjornharrtell** [comment]
I've just made a 1.6.2 release without an empty index.js module. Not sure if it will help or if it will cause other issues and would appreciate any feedback from testing it with webpack and elsewhere.

**Comment by bjornharrtell** [comment]
And now I've also released 2.0.0 which is a large refactoring to ES6 classes.
